tests/posix/xsi_threads_ext/src/main.c

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2024, Meta
+ * Copyright (c) 2024, Marvin Ouma <pancakesdeath@protonmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <pthread.h>
+
+#include <zephyr/sys/util.h>
+#include <zephyr/ztest.h>
+
+#define BIOS_FOOD 0xB105F00D
+
+static bool attr_valid;
+static pthread_attr_t attr;
+static const pthread_attr_t uninit_attr;
+static bool detached_thread_has_finished;
+
+/* TODO: this should be optional */
+#define STATIC_THREAD_STACK_SIZE (MAX(1024, PTHREAD_STACK_MIN + CONFIG_TEST_EXTRA_STACK_SIZE))
```

Comment:
Where is `CONFIG_TEST_EXTRA_STACK_SIZE` set in the `common` test suite?

Reply:
It has been set in both https://github.com/zephyrproject-rtos/zephyr/blob/5dfd41e51a86c75d156241b7566ad4c1c1dce2d5/tests/posix/common/testcase.yaml#L71 and https://github.com/zephyrproject-rtos/zephyr/blob/5dfd41e51a86c75d156241b7566ad4c1c1dce2d5/tests/posix/common/prj.conf#L13. 
I will set it.

Reply:
It was added here, and likely completely unnecessary for this test

https://github.com/zephyrproject-rtos/zephyr/commit/43f917111b30790801824cab050a758817fd03f2